subprojects/platform-play/src/test/groovy/org/gradle/play/internal/PlayPlatformResolverTest.groovy

```diff
@@ -44,49 +44,38 @@ class PlayPlatformResolverTest extends Specification {
 
         then:
         def e = thrown(InvalidUserDataException)
-        e.message == "Not a supported Play version: 2.1.0. This plugin is compatible with: [2.4.x, 2.3.x, 2.2.x]."
+        e.message == "Not a supported Play version: 2.1.0. This plugin is compatible with: [2.5.x, 2.4.x, 2.3.x, 2.2.x]."
 
         where:
         requirement << ["play-2.1.0", [play: '2.1.0']]
     }
 
-    def "resolves platform for Play 2.2.x"() {
+    def "resolves platform for Play"() {
         when:
         def playPlatform = resolve(requirement)
 
         then:
-        playPlatform.name == "play-2.2.3"
-        playPlatform.playVersion == "2.2.3"
+        playPlatform.name == requirement
+        playPlatform.playVersion == playVersion
         playPlatform.javaPlatform.targetCompatibility == JavaVersion.current()
-        playPlatform.scalaPlatform.scalaVersion == "2.10.4"
+        playPlatform.scalaPlatform.scalaVersion == scalaPlatform
 
         where:
-        requirement << ["play-2.2.3", [play: '2.2.3']]
+        requirement       | playVersion | scalaPlatform
+        "play-2.2.3"      | "2.2.3"     | "2.10.4"
+        "play-2.3.4"      | "2.3.4"     | "2.11.4"
+        "play-2.4.6"      | "2.4.6"     | "2.11.4"
+        "play-2.5.4"      | "2.5.4"     | "2.11.4"
+        "play-2.3.1-2.10" | "2.3.1"     | "2.10.4"
```

Comment:
This test fails for `play-2.3.1-2.10` - please fix it. Requirement should be - as before - sometimes a `Map` and sometimes a `String`. If it gets to complicated feel free to split this tests to multiple tests again.